docs/articles/documentation/test-api/identify-the-browser-and-platform.md

```diff
@@ -0,0 +1,289 @@
+
+```js
+{
+    prettyUserAgent: 'Firefox 69.0 / Windows 10'
+}
+```
+
+```js
+{
+    prettyUserAgent: 'Chrome 77.0.3865.120 / macOS 10.15.1'
+}
+```
+
+## Examples
+
+### Create a Browser-Specific Hook
+
+The following example shows how to create a [beforeEach](test-code-structure.md#test-hooks) hook that runs for only particular [browser engines](#engine).
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://example.com`
+    .beforeEach(async t => {
+        if (t.browser.engine.name === 'Blink')
+            return;
+
+        // ...
+    });
+```
+
+### Generate the Screenshot File Path
+
+This example shows how to generate the screenshot path based on the [browser name](#name). This prevents screenshots taken with [t.takeElementScreenshot](actions/take-screenshot.md#take-a-screenshot-of-a-page-element) in different browsers from being overwritten.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://example.com`;
+
+test('My test', async t => {
+    const loginButton = Selector('div').withText('Login');
+
+    await t.takeElementScreenshot(loginButton, `auth/${t.browser.name}/login-button.png`);
+});
+```
+
+### Verify the Installer Version
+
+The following example verifies that the website detects the user's platform correctly and offers to download the right installer. This test uses the [t.browser.os.name](#os) property to determine the operating system and check the installer name.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://example.com`;
+
+const downloadButton = Selector('a').withText('Download');
+
+test('Check the installer platform', async t => {
+    let installerName;
+
+    switch (t.browser.os.name) {
+        case 'Windows':
+            installerName = 'myapp-win.exe';
+            break;
+        case 'macOS':
+            installerName = 'myapp-mac.dmg';
+            break;
+        default:
+            installerName = 'myapp-linux.tar.gz';
+            break;
+    }
+
+    await t.expect(downloadButton.getAttribute('href')).contains(installerName);
+});
+```
+
+### Block Ads on Mobile Devices
+
+The following example uses the [t.browser.platform](#platform) property to determine the platform and [attach](intercepting-http-requests/attaching-hooks-to-tests-and-fixtures.md) a [request mock](intercepting-http-requests/mocking-http-requests.md) that blocks ad banners if the test runs on a mobile device.
+
+```js
+import { RequestMock } from 'testcafe';
+
+const mobileAdMock = RequestMock()
+    .onRequestTo(/bannernetwork.com/)
+    .respond((req, res) => { res.statusCode = '404'; });
+
+fixture `My fixture`
+    .beforeEach(async t => {
+        if (t.browser.platform === 'mobile')
+            await t.addRequestHooks(mobileAdMock);
+
+        await t.navigateTo('https://mysite.com/tested/page/');
```

Comment:
I'm not sure it makes sense. For the example purposes, we assume that blocking requests is enough to disable ads.

In this case, it seems like we are doing the same thing twice. Even if it makes sense in practice, it's not obvious in case of a synthetic example.